src/azure-cli/azure/cli/command_modules/resource/custom.py

```diff
@@ -1947,6 +1949,10 @@ def unregister_provider(cmd, resource_provider_namespace, wait=False):
     _update_provider(cmd.cli_ctx, resource_provider_namespace, registering=False, wait=wait)
 
 
+def register_management_group(cmd, resource_provider_namespace, management_group_id, wait=False):
+    _update_provider(cmd.cli_ctx, resource_provider_namespace, registering=True, wait=wait, mg=True, mgID=management_group_id)
```

Comment:
Hi @tjegbejimba, because we could use one function for registering subscription and management group, could we consider a new design using `az provider register` with `--mg` instead of `az provider register-mg`?
utilizing `az provider register` would be more extensible